app/src/main/java/com/omkarmoghe/pokemap/views/LoginActivity.java

```diff
@@ -48,13 +50,22 @@
     private GoogleManager.LoginListener mGoogleLoginListener;
 
     private String mDeviceCode;
+    private PokemapAppPreferences mPref;
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
-        setContentView(R.layout.activity_login);
+
         mNianticManager = NianticManager.getInstance();
         mGoogleManager = GoogleManager.getInstance();
+        mPref = new PokemapSharedPreferences(this);
+        
+        if (mPref.isUsernameSet() || mPref.isPasswordSet()) {
```

Comment:
Might want to make this an and (mPref.isUsernameSet() && mPref.isPasswordSet()). Of course, this shouldn't really matter much for dev purposes.